Modules/ITS/src/ITSChipStatusTask.cxx

```diff
@@ -0,0 +1,178 @@
+      msg->SetTextSize(20);
+      msg->SetTextFont(43);
+      DeadChips[i]->GetListOfFunctions()->Add(msg);
+      if (iLayer < iLayerEnd) {
+        auto l = new TLine(0, ChipBoundary[iLayer + 1] - 0.5, nQCCycleToMonitor, ChipBoundary[iLayer + 1] - 0.5);
+        DeadChips[i]->GetListOfFunctions()->Add(l);
+      }
+    }
+    int nChipsPerStave = i == 0 ? 9 : i == 1 ? 112
+                                             : 196;
+    for (int iy = 0; iy < DeadChips[i]->GetNbinsY(); iy++) {
+      if (iy % nChipsPerStave == 0) {
+        auto l = new TLine(0, iy - 0.5, nQCCycleToMonitor, iy - 0.5);
+        l->SetLineColor(40);
+        l->SetLineWidth(1);
+
+        DeadChips[i]->GetListOfFunctions()->Add(l);
+
+        if (iy >= ChipBoundary[Layer_Draw + 1]) {
+          Layer_Draw++;
+        }
+        TLatex* msg = new TLatex(-3, iy, Form("%d", (int)(iy) / nChipsPerStave - StaveBoundary[Layer_Draw]));
+        msg->SetTextSize(12);
+        msg->SetTextFont(43);
+        DeadChips[i]->GetListOfFunctions()->Add(msg);
+      }
+    }
+    Layer_Draw++;
+
+    DeadChips[i]->GetYaxis()->SetLabelSize(0);
+    DeadChips[i]->GetYaxis()->SetAxisColor(0);
+    getObjectsManager()->startPublishing(DeadChips[i]);
+  }
+}
+
+void ITSChipStatusTask::setAxisTitle(TH1* object, const char* xTitle, const char* yTitle)
+{
+  object->GetXaxis()->SetTitle(xTitle);
+  object->GetYaxis()->SetTitle(yTitle);
+}
+
+void ITSChipStatusTask::startOfActivity(const Activity& activity)
+{
+  ILOG(Debug, Devel) << "startOfActivity : " << activity.mId << ENDM;
+}
+
+void ITSChipStatusTask::startOfCycle()
+{
+  ILOG(Debug, Devel) << "startOfCycle" << ENDM;
+
+  for (int i = 0; i < 3; i++) {
+    CurrentDeadChips[i].clear();
+    CurrentDeadChips[i].resize(ChipsBoundaryBarrels[i + 1] - ChipsBoundaryBarrels[i], 0);
+  }
+}
+
+void ITSChipStatusTask::monitorData(o2::framework::ProcessingContext& ctx)
+{
+  auto aliveChips = ctx.inputs().get<gsl::span<char>>("chipstatus");
+
+  int id = 0;
+  for (auto chipID : aliveChips) {
+    int iBarrel = id < ChipsBoundaryBarrels[1] ? 0 : id < ChipsBoundaryBarrels[2] ? 1
+                                                                                  : 2;
+    if ((int)chipID != 1) {
+      int ChipID_fill = id - ChipsBoundaryBarrels[iBarrel];
+      if ((ChipID_fill < 0) || (ChipID_fill >= CurrentDeadChips[iBarrel].size())) {
+        ILOG(Warning, Devel) << " prolematic ID for chip: " << ChipID_fill << ENDM;
+      } else {
+        CurrentDeadChips[iBarrel][ChipID_fill]++;
+      }
+    }
+    id++;
+  }
+}
+
+void ITSChipStatusTask::getParameters()
+{
+
+  nQCCycleToMonitor = o2::quality_control_modules::common::getFromConfig<int>(mCustomParameters, "nQCCycleToMonitor", nQCCycleToMonitor);
+}
+
+void ITSChipStatusTask::endOfCycle()
+{
+
+  ILOG(Debug, Devel) << "endOfCycle" << ENDM;
+  for (int iBarrel = 0; iBarrel < 3; iBarrel++) {
+    ChipsStack[iBarrel]->push(CurrentDeadChips[iBarrel]);
```

Comment:
A question on `push`. If I understand well, if we reached a QC cycle number which is > `nQCCycleToMonitor`, you call `std::rotate` to shift all elements up by 1 position and then you store the new element at the end. But after that you fill the TH2D meaning that if before you had a vector of dead chips at QCcycle = i, now the same vector is at QCcycle = i-1 which means that the TH2D will have the same entry twice, once at x=i and then at x=i-1. Probably I'm missing something. Thanks.

Reply:
You are right, but when I call rotate() function, I use not push_back() but just back() command, which fixes the problem you mentioned